### ISSUE TITLE
embedding_outside not used in Skipgram.forward() (01 - Word2Vec (Skipgram).ipynb)

### ISSUE THREAD

**Issue by MohammadElsharqawy** [chaklam-silpasuwanchai/Natural-Language-Processing#6]
Hi! 👋

In the notebook 01 - Word2Vec (Skipgram).ipynb, the Skipgram class defines two separate embedding layers:

self.embedding_center = nn.Embedding(voc_size, emb_size)
self.embedding_outside = nn.Embedding(voc_size, emb_size)

However, in the forward() method, only embedding_center is used — even for the outside words and vocab logits:
```
outside_embedding = self.embedding_center(outside)
all_vocabs_embedding = self.embedding_center(all_vocabs)
```

It seems this might just be a small typo. Should these lines instead use self.embedding_outside? Like this:
```
outside_embedding = self.embedding_outside(outside)
all_vocabs_embedding = self.embedding_outside(all_vocabs)
```

Thanks so much for the great notebook! Really enjoying it. 🚀

**Comment by chaklam-silpasuwanchai** [comment]
Thanks for the heads up.  You might be correct!   I will fix it.Bests,Chaklam On 19 Apr BE 2568, at 05:52, Mohammad Elsharqawy ***@***.***> wrote:﻿
  Hi! 👋
In the notebook 01 - Word2Vec (Skipgram).ipynb, the Skipgram class defines two separate embedding layers:
self.embedding_center = nn.Embedding(voc_size, emb_size)
self.embedding_outside = nn.Embedding(voc_size, emb_size)
However, in the forward() method, only embedding_center is used — even for the outside words and vocab logits:
outside_embedding = self.embedding_center(outside)
all_vocabs_embedding = self.embedding_center(all_vocabs)

It seems this might just be a small typo. Should these lines instead use self.embedding_outside? Like this:
outside_embedding = self.embedding_outside(outside)
all_vocabs_embedding = self.embedding_outside(all_vocabs)

Thanks so much for the great notebook! Really enjoying it. 🚀—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>



MohammadElsharqawy created an issue (chaklam-silpasuwanchai/Python-fo-Natural-Language-Processing#6)
Hi! 👋
In the notebook 01 - Word2Vec (Skipgram).ipynb, the Skipgram class defines two separate embedding layers:
self.embedding_center = nn.Embedding(voc_size, emb_size)
self.embedding_outside = nn.Embedding(voc_size, emb_size)
However, in the forward() method, only embedding_center is used — even for the outside words and vocab logits:
outside_embedding = self.embedding_center(outside)
all_vocabs_embedding = self.embedding_center(all_vocabs)

It seems this might just be a small typo. Should these lines instead use self.embedding_outside? Like this:
outside_embedding = self.embedding_outside(outside)
all_vocabs_embedding = self.embedding_outside(all_vocabs)

Thanks so much for the great notebook! Really enjoying it. 🚀
